app/routes/reorder/index.js

```diff
@@ -0,0 +1,187 @@
+import * as React from "react";
+import { json } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+
+import Instructions from "./_components/Instructions.mdx";
+import Track from "./_components/Track.js";
+import { getTracks } from "./_data-sources/track.server.js";
+
+import styles from "./styles/index.css";
+
+export const loader = async () => json(await getTracks());
+
+export const links = () => {
+  return [{ rel: "stylesheet", href: styles }];
+};
+
+export default function ReorderRoute() {
+  const {
+    tracks,
+    focusTrack,
+    setFocusTrack,
+    focusWithinTrack,
+    clickFocusWithinTrack,
+    focusRef,
+    onKeyDown,
+  } = useSetup();
+
+  return (
+    <main>
+      <Instructions />
+      <a href=".">Random link to focus on</a>
+      <h2>Primary Tracks</h2>
+      <ol className="tracks">
+        {tracks.map((track) => {
+          const hasFocus = focusTrack === track.id;
+          return (
+            <Track
+              key={track.id}
+              tabIndex={hasFocus ? 0 : -1}
+              ref={hasFocus ? focusRef : null}
+              onClick={(e) => {
+                setFocusTrack(e.currentTarget.id);
+              }}
+              {...{
+                track,
+                focusWithinTrack,
+                clickFocusWithinTrack,
+                onKeyDown,
+              }}
+            />
+          );
+        })}
+      </ol>
+      <button>Random button to focus on</button>
+    </main>
+  );
+}
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "set-track": {
+      const focusTrack =
+        typeof action.value === "function" ? action.value(state) : action.value;
+
+      return { focusTrack, focusWithinTrack: null };
+    }
+
+    case "set-within": {
+      return { ...state, focusWithinTrack: action.value };
+    }
+
+    case "click-within": {
+      return {
+        focusTrack: action.value.track,
+        focusWithinTrack: action.value.button,
+      };
+    }
+
+    default:
+      throw new Error(`Invalid action ${action.type}`);
+  }
+};
+
+const FOCUS_WITHIN_FSM = new Map([
```

Comment:
Simple finite state machine FTW. This is used to track what to focus on within a track.